### PR TITLE
Applied structured data to Dynamics Cog

### PR DESCRIPTION
**Pull request by saludangelito**

### CODE REVIEW

File: src/steps/lead-field-equals.ts [hunk 1]
```diff
@@ -46,25 +46,24 @@ export class LeadFieldEquals extends BaseStep implements StepInterface {
```
Comment (iamEAP) [inline src/steps/lead-field-equals.ts]: - [ ] Would also be useful to add the `expectedRecords` property to this step too, including the guaranteed fields recommended for the create step above.

File: src/steps/delete-lead.ts [hunk 2]
```diff
@@ -34,7 +34,9 @@ export class DeleteLead extends BaseStep implements StepInterface {
         };
         const result = await this.client.delete(deleteRequest);
         if (result) {
```
[the remark this answers is not in the document]
Comment (saludangelito) [reply src/steps/delete-lead.ts]: upon checking again while working on this task, `isDeleted` returns just a boolean which so far always returns true. Errors are returned as an...error... thats caught in the catch block when the leadId provided for delete is either wrong format or just happens that the lead doesnt exist. So i thought the extra complexity isn't necessary.

[the remark this answers is not in the document]
Comment (saludangelito) [reply src/steps/delete-lead.ts]: Now that i think about it maybe i should also remove the one in the step as well...

File: src/steps/lead-field-equals.ts [hunk 3]
```diff
@@ -46,25 +46,24 @@ export class LeadFieldEquals extends BaseStep implements StepInterface {
 
     try {
       const records = await this.client.retrieveMultiple(request);
```
[the remark this answers is not in the document]
Comment (saludangelito) [reply src/steps/lead-field-equals.ts]: Turned out i just had to remove the select query/property

File: src/steps/delete-lead.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (iamEAP) [reply src/steps/delete-lead.ts]: 👌

**Pull request merged**